tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import slack
+import os
+import json
+import requests
+import time
+import re
+import logging
+from pathlib import Path
+from dotenv import load_dotenv
+from flask import Flask, request, Response
+from multiprocessing import Process
+from googleapiclient.discovery import build_from_document
+from datetime import datetime
+
+# To run this on the cheapest possible VM, we will only log Warnings and Errors
+logging.basicConfig(filename='error.log')
+flask_logger = logging.getLogger('werkzeug')
+flask_logger.setLevel(logging.WARNING)
+
+logging.warning('Started at: {}'.format(datetime.now()))
+
+# Load our environment variables from the .env file
+env_path = Path('.') / '.env'
+load_dotenv(dotenv_path=env_path)
+
+app = Flask(__name__)
+
+client = slack.WebClient(token=os.environ['SLACK_TOKEN'])
+ORG_ID = os.environ['ORG_ID']
+API_KEY = os.environ['API_KEY']
+
+discovery_doc_file = 'cloudsupport.v2alpha.json'
+
+# Get our discovery doc
+r = requests.get('https://cloudsupport.googleapis.com/$discovery/rest?key={}&labels=V2_TRUSTED_TESTER&version=v2alpha'.format(API_KEY))
+
+# Add the discovery doc, or replace the previous version with the latest one
+if os.path.exists(discovery_doc_file):
+    os.remove(discovery_doc_file)
+with open(discovery_doc_file, "w") as f:
+    json.dump(r.json(), f)
+
+with open(discovery_doc_file) as f:
+    dicovery_doc = f.read()
+support_service = build_from_document(dicovery_doc)
+
+cases_file = 'support_cases.json'
+tracked_cases_file = 'tracked_cases'
+
+if os.path.exists(tracked_cases_file):
+    with open(tracked_cases_file) as tcf:
+        tracked_cases = list(eval(tcf.read()))
+else:
+    tracked_cases = []
+
```

Comment:
Remove extra blank line (2 lines of separation before class, as per PEP8 https://www.python.org/dev/peps/pep-0008/)